analyzer/src/funTest/assets/projects/synthetic/pub-expected-output-project-with-flutter.yml

```diff
@@ -12750,6 +12750,35 @@ packages:
       revision: "master"
       path: "pkg/analyzer"
   curations: []
+- package:
+    id: "Pub:archive:archive:2.0.11"
```

Comment:
Why was this package added to the packages list, but not to the dependency tree? Same question applies to other packages added below.

Reply:
ORTs `Pub` implementation has the recursive `buildDependencyTree()`. It skips all packages which have the attribute `source = sdk` set in the `lockfile` since the implementation cannot parse such packages. As the only `dependant` of `archive` is `flutter_test` which is skipped there is no `(package) reference` to `archive`. Similar for other packages.

As the issue existed already before, e.g. the `flutter` package was not referenced, can we consider it out of scope of this PR?

I would actually volunteer to fix that issue soon, but preferably after my current main task.

Reply:
Yes, I checked and we already have packages like `Pub:flutter:flutter` which are in the package list but not in the dependency tree. Can you create an issue for this, please? We need to make a decision if they should be removed from the package list or added to the tree.

Reply:
> Can you create an issue for this, please?

BTW, you can easily do this from your own comment by clicking the three dots, and then "Reference in new issue".

Reply:
Sure, creating an issue now...